index.js

```diff
@@ -102,8 +103,17 @@ class HandlebarsPlugin {
         };
 
         if (compiler.hooks) {
-            compiler.hooks.make.tapAsync("HandlebarsRenderPlugin", compile);
-            compiler.hooks.emit.tapAsync("HandlebarsRenderPlugin", emitDependencies);
+            // tap into the webpack html plugin hooks
+            if(this.options.htmlWebpackPlugin) {
+                compiler.hooks.afterEmit.tap('HandlebarsRenderPlugin', compilation => {
```

Comment:
If I change `compiler.hooks.afterEmit` to `compiler.hooks.make` things start to work better.

Reply:
Wow :open_mouth:  . Thanks for all the digging. Getting things to work with webpack-dev-server was a second step for me, but ultimately would be awesome.

> Another option might be to get the html-contents directly, before it is saved to disk

Getting the html in a callback from html-webpack-plugin and write the partial hbs ourselves?

Reply:
Ah sorry. The webpack-dev-server is a _must_. I had many issues on this and its now running pretty well.